lib/src/from_handlers.dart

```diff
@@ -67,12 +67,9 @@ class _StreamTransformer<S, T> implements StreamTransformer<S, T> {
       controller.onResume = () => subscription?.resume();
     }
     controller.onCancel = () {
-      if (controller.hasListener || subscription == null) {
-        return new Future.value();
-      }
-      var toCancel = subscription;
+      if (controller.hasListener || subscription == null) return;
+      subscription.cancel();
```

Comment:
The `subscription.cancel` call *can* return a future – it comes from the `Stream` you took as argument, so you have no idea whether it returns a future or not – and that future can contain an error, so you *should* forward it so the error can be properly handled.